lib/matplotlib/axes/_axes.py

```diff
@@ -4507,6 +4507,11 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
             x, y, s, c, colors, edgecolors, linewidths = \
                 cbook._combine_masks(
                     x, y, s, c, colors, edgecolors, linewidths)
+        # Unmask edgecolors if it was actually a single RGB or RGBA.
+        if (x.size in (3, 4)
```

Comment:
Do we need a similar block for `colors` as well?

Reply:
I believe that is already an (N,4) array out of `_parse_scatter_color_args` and should have broadcast correctly with the masks.

Reply:
No, because the case that was problematic for colors was the case where edgecolors was being
set from colors, above all this.  So handling edgecolors here takes care of it.